contracts/strategies/CurveStrategy.sol

```diff
@@ -199,12 +221,19 @@ contract CurveStrategy is Ownable, IStrategy {
             if (uint128(_inboundTokenIndex) >= NUM_AAVE_TOKENS) {
                 revert INVALID_INBOUND_TOKEN_INDEX();
             }
-            lpToken = IERC20(pool.lp_token());
+            lpToken = IERC20(_pool.lp_token());
         } else {
-            if (uint128(_inboundTokenIndex) >= NUM_ATRI_CRYPTO_TOKENS) {
-                revert INVALID_INBOUND_TOKEN_INDEX();
+            if (_poolType == ATRI_CRYPTO_POOL) {
```

Comment:
for matic & atricrypto type of pools the method to get the lp token is same so hence I have this together let me know wdyt

Reply:
my suggestion is to optimize this constructor as below. the `ifs` blocks can be considerably reduced.
looks like that `lpToken = IERC20(_pool.token());` is always set in the same way, no matter the type of pool, so it can be placed out of the `ifs`.

Below is a suggestion for the whole constructor (so it's easier to spot the whole logic):

```
    constructor(
        ICurvePool _pool,
        int128 _inboundTokenIndex,
        uint64 _poolType,
        ICurveGauge _gauge,
        IERC20 _rewardToken,
        IERC20 _curve,
        ICurveMinter _gaugeMinter
    ) {
        if (address(_pool) == address(0)) {
            revert INVALID_POOL();
        }
        if (address(_gauge) == address(0)) {
            revert INVALID_GAUGE();
        }
        if (address(_curve) == address(0)) {
            revert INVALID_CURVE_TOKEN();
        }
        if (address(_rewardToken) == address(0)) {
            revert INVALID_REWARD_TOKEN();
        }

        if (_poolType > MATIC_POOL) {
            revert INVALID_POOL_TYPE();
        }

        if (
            (_inboundTokenIndex < 0)
            || (_poolType == AAVE_POOL && uint128(_inboundTokenIndex) >= NUM_AAVE_TOKENS)
            || (_poolType == ATRI_CRYPTO_POOL && uint128(_inboundTokenIndex) >= NUM_ATRI_CRYPTO_TOKENS)
            || (_poolType == MATIC_POOL && uint128(_inboundTokenIndex) >= NUM_MATIC_POOL_TOKENS)
        ) {
            revert INVALID_INBOUND_TOKEN_INDEX();
        }

        pool = _pool;
        gauge = _gauge;
        gaugeMinter = _gaugeMinter;
        curve = _curve;
        poolType = _poolType;
        inboundTokenIndex = _inboundTokenIndex;
        // wmatic in case of polygon and address(0) for non-polygon deployment
        rewardToken = _rewardToken;
        lpToken = IERC20(_pool.token());
    }
```
 
Note I also added the following new validation -  This ensures `_poolType` is a correct input as well:
```
         if (_poolType > MATIC_POOL) {
            revert INVALID_POOL_TYPE();
        }
```

 Also note we can combine the condition `(_inboundTokenIndex < 0)` in the `if` block that throws the `INVALID_INBOUND_TOKEN_INDEX` error.